src/Core/Configurations/ConfigFileWatcher.cs

```diff
@@ -38,7 +38,8 @@ private void OnConfigFileChange(object sender, FileSystemEventArgs e)
 
             if (!_configProvider.IsLateConfigured && _configProvider.GetConfig().IsDevelopmentMode())
             {
-                _configProvider.HotReloadConfig();
+                // pass along the original default data source name for consistency within runtime config's data structures
+                _configProvider.HotReloadConfig(_configProvider.GetConfig().DefaultDataSourceName);
```

Comment:
Why do you need to pass along this from the ConfigFileWatcher when you call HotReloadConfig? Doesnt _configProvider already have reference to the old config? So, can't you simply do GetConfig().DefaultDataSourceName inside the configProvider.HotReloadConfig function?

Reply:
this is the case, so I will just pass along the DefaultDataSourceName from the reference we already have available in the config provider.